src/app/gui/queryresults/queryresultsservice.js

```diff
@@ -1341,99 +1697,112 @@ proto.printAtlas = function(layer, feature){
+                : [features]
+              : features;
+  const {
+    query = {}
+  } = this.state;
+  const fids = features.map(feature => feature.attributes[G3W_FID]).join(',');
+  const data = { fids };
+
   /**
-   * is a function that che be called in case of querybypolygon
+   * A function that che be called in case of querybypolygon
+   * 
    * @param active
    */
   const runDownload = async (active=false) => {
     if (features.length > 1) {
       layer[DownloadFormats.name].active = active;
-      this.setLayerActionTool({
-        layer
-      })
+      this.setLayerActionTool({ layer });
     }
-    const projectLayer = CatalogLayersStoresRegistry.getLayerById(layer.id);
     const download_caller_id = ApplicationService.setDownload(true);
     GUI.setLoadingContent(true);
     try {
-      await projectLayer.getDownloadFilefromDownloadDataType(type, {
-        data
-      }) || Promise.resolve();
-    } catch(err){
+      await CatalogLayersStoresRegistry.getLayerById(layer.id).getDownloadFilefromDownloadDataType(type, { data }) || Promise.resolve();
+    } catch(err) {
       GUI.notify.error(err || t("info.server_error"));
     }
     ApplicationService.setDownload(false, download_caller_id);
     GUI.setLoadingContent(false);
-    const downloadsactions = this.state.layersactions[layer.id].find(action => action.id === 'downloads');
-    if (features.length > 1) {
-      if (downloadsactions === undefined) {
-        layer[type].active = false;
-        this.setLayerActionTool({
-          layer
-        })
-      }
-      else layer[DownloadFormats.name].active = false;
+    const downloadsactions = this.state.layersactions[layer.id].find(action => 'downloads' === action.id);
+
+    /** @FIXME add description */
+    if (features.length > 1 && undefined === downloadsactions) {
+      layer[type].active = false;
+      this.setLayerActionTool({ layer });
     }
-    else {
-      if (downloadsactions === undefined) action.state.toggled[index] = false;
-      else downloadsactions.state.toggled[index] = false;
-      this.setCurrentActionLayerFeatureTool({
-        index,
-        action,
-        layer
-      });
+
+    /** @FIXME add description */
+    if(features.length > 1 && undefined !== downloadsactions) {
+      layer[DownloadFormats.name].active = false;
+    }
+
+    /** @FIXME add description */
+    if (features.length <= 1 && undefined === downloadsactions) {
+      action.state.toggled[index] = false;
+    }
+
+    /** @FIXME add description */
+    if (features.length <= 1 && undefined !== downloadsactions) {
+      downloadsactions.state.toggled[index] = false;
+    }
+
+    /** @FIXME add description */
+    if (features.length <= 1) {
+      this.setCurrentActionLayerFeatureTool({ index, action, layer });
     }
   };
-  if (query.type === 'polygon'){
-    //check if multidownload if present
+
+  if ('polygon' !== query.type) {
+    runDownload();
+  } else { // check if multi-download if present
```

Comment:
within this function `QueryResultsService::downloadFeatures(type, layer, features=[], action, index)`, i rewrote several if-else conditions to reduce nesting level, check carefully